main.tf

```diff
@@ -65,7 +70,7 @@ resource "azurerm_network_interface" "dynamic" {
 resource "azurerm_linux_virtual_machine" "linux" {
   count = (var.kernel_type == "linux" ? 1 : 0)
 
-  name                = local.name
+  name                = local.virtual_machine_name

```

Comment:
@jhisc in my mind I had thought to keep the resource name the same and simply edit the hostname with the computer_name option that is present for both linux and windows VMs. But this gets the same result and keeps the name the same in the portal

Reply:
Personally I think we should match up the portal name with the hostname.
Will probably make life a lot easier than having to map the two.

We have all the core information in tags anyway